mender-auth/ipc/client.hpp

```diff
@@ -0,0 +1,121 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+
+#ifndef MENDER_AUTH_CLIENT_HPP
+#define MENDER_AUTH_CLIENT_HPP
+
+#include <functional>
+#include <string>
+
+#include <common/conf.hpp>
+#include <common/error.hpp>
+#include <common/events.hpp>
+#include <common/http.hpp>
+
+#include <common/http.hpp>
+#include <api/api.hpp>
+#include <api/auth.hpp>
+
+namespace mender {
+namespace ipc {
+
+
+using namespace std;
+
+namespace auth_client = mender::api::auth;
+namespace conf = mender::common::conf;
+namespace error = mender::common::error;
+namespace events = mender::common::events;
+namespace mlog = mender::common::log;
+
+
+enum IPCErrorCode {
+	NoError = 0,
+	SetupError,
+	RequestError,
+	ResponseError,
+	APIError,
+	UnauthorizedError,
+};
+
+class IPCErrorCategoryClass : public std::error_category {
+public:
+	const char *name() const noexcept override;
+	string message(int code) const override;
+};
+extern const IPCErrorCategoryClass IPCErrorCategory;
+
+error::Error MakeError(IPCErrorCode code, const string &msg);
+
+namespace type {
+namespace webserver {
+
+class Caching {
+public:
+	error::Error ListenForEvents(
+		events::EventLoop &loop,
+		http::Server &server,
+		const string &server_url,
+		http::Client &client,
+		const conf::MenderConfig &config,
+		const string &private_key_path,
+		const string &identity_script_path);
+
+	void CacheAPIResponse(const string &server_url, auth_client::APIResponse resp) {
```

Comment:
AFAICS, this should only be called from within the handler. Should it be private?